### PR TITLE
VM usb nic passthrough (from Incus)

### PR DESCRIPTION
**Pull request by MggMuggins**
What constitutes adequate testing for this set of changes? I don't have access (currently) to a USB/Ethernet adapter; while I'm sure it's _possible_ to simulate one I doubt that would be a straightforward/worthwhile process. Should I check if someone else on the team has appropriate hardware to run a quick check on this?

### CODE REVIEW

**Comment by simondeziel** [comment]
> What constitutes adequate testing for this set of changes? I don't have access (currently) to a USB/Ethernet adapter;

When I plug my iphone to my laptop, an `AppleUSBEthernet` NIC appears, maybe you could use your phone for a local test? I could also do the test for you if you need.

> while I'm sure it's _possible_ to simulate one I doubt that would be a straightforward/worthwhile process.

Yeah, last I checked, simulating one was not easy which is why we still have no USB test part of `lxd-ci` :/

> Should I check if someone else on the team has appropriate hardware to run a quick check on this?

If the phone trick doesn't work, I also have cheap USB Wifi NICs laying around so let me know.

**Comment by MggMuggins** [comment]
Had to enable tethering on my phone but that works on Android, I'll see what I can do. Thanks.

**Comment by MggMuggins** [comment]
I sideloaded a `lxd.debug` into `latest/edge` and was able to pass my phone through to the VM. This is good to go!

File: lxd/instance/instancetype/instance.go [hunk 1]
```diff
@@ -1210,6 +1210,14 @@ func ConfigKeyChecker(key string, instanceType Type) (func(value string) error,
 			return validate.IsAny, nil
 		}
 
+		if strings.HasSuffix(key, ".bus") {
```
Comment (tomponline) [inline lxd/instance/instancetype/instance.go:1213]: Please can you add the lxdmeta:generate documentation for these 2 new keys. Thanks

**Review by tomponline — CHANGES_REQUESTED** [review]
Looks good thanks.

Just needs the new volatile keys documenting.

**Comment by MggMuggins** [comment]
Added the docstrings; it doesn't look like this combination of `entities=network-physical` and `group=volatile` is being used anywhere in the docs. The [volatile instance keys](https://documentation.ubuntu.com/lxd/en/latest/reference/instance_options/#volatile-internal-data) have a section dedicated, but the nic option docs are [nested pretty deeply](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_nic/#nictype-physical), so I'm not sure if it makes sense to include the volatile keys. @ru-fu do you have memory/opinion on this?

**Review by tomponline — APPROVED** [review]

**Comment by ru-fu** [comment]
> Added the docstrings; it doesn't look like this combination of `entities=network-physical` and `group=volatile` is being used anywhere in the docs. The [volatile instance keys](https://documentation.ubuntu.com/lxd/en/latest/reference/instance_options/#volatile-internal-data) have a section dedicated, but the nic option docs are [nested pretty deeply](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_nic/#nictype-physical), so I'm not sure if it makes sense to include the volatile keys. @ru-fu do you have memory of previous discussion/an opinion on this?

No previous discussions about this.
My suggestion would be to just add the keys to the common `network-conf` group (and maybe add a sentence to the description that they are internal and shouldn't be changed).

In general, we break up the set of config options when there are a lot and "it makes sense", but there isn't a strict rule on when that needs to happen. And in this case, they will appear at the end of the list anyway, so I think it's totally fine to add them to the list [here](https://documentation.ubuntu.com/lxd/en/latest/reference/network_physical/#configuration-options).

**Pull request merged**